### ISSUE TITLE
Embedded class is incorrectly recognized as `Unknown` under particular conditions

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#2157]
## Environment data

-   Language Server version: 2021.12.2-pre.1
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.12 Anaconda

## Sample code

```python
from __future__ import annotations

def b0() -> A.B: ...
def c0() -> A.B.C: ...

class A:
    class B:
        class C:
            ...
    def b(self) -> A.B: ...
    def c(self) -> A.B.C: ...

def b1() -> A.B: ...
def c1() -> A.B.C: ...

# _ = b0()
# _ = c0()
# _ = A().b()
# _ = A().c()
```

## Expected behavior

Every type is correctly recognized.

## Actual behavior

When you hover on an embedded part (e.g. `B` or `C` in `A.B.C`) of the result types of `b0`, `c0`, `A.b` and `A.c`, you will find that it is recognized as `Unknown`.

`b1` and `c1` are OK.

~~All become right once the last lines are recovered from being commented out.~~
For the last several commented lines, if you uncomment a line, the result type of the corresponding function / method SOMETIMES becomes right, while I didn't find a way to ensure that.

And to my surprise, this issue only exists for return types, but not argument types.

Edit:
Change the description of uncommenting the last several lines.

**Comment by erictraut** [comment]
Thanks for the bug report. This occurs only when `from __future__ import annotations` is in effect and the return type annotation refers to a forward-declared symbol and the type of the annotation is evaluated prior to the forward-declared symbol by Pyright's lazy type evaluator.

This will be fixed in the next release.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.12.1, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021121-9-december-2021)

**Comment by Azureblade3808** [comment]
Doesn't seem to get fixed in 2021.12.1 | 2021.12.2-pre.1.

**Comment by Azureblade3808** [comment]
> This issue has been fixed in version 2021.12.1, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021121-9-december-2021)

Is the fix not published yet? For this issue is found in 2021.12.1 and I haven't see a new version of 2021.12.2 in case there's a typo.

**Comment by erictraut** [comment]
This was fixed after Pylance 2021.12.1 was released.

**Comment by Azureblade3808** [comment]
Are you talking about 2021.12.2 instead of 2021.12.1?

2021.12.1 is just the version of Pylance which the issue was found in.

**Comment by erictraut** [comment]
@Azureblade3808, the fix will be in Pylance version 2021.12.2 which will be released next week. (Typically we publish Pylance on Wednesday or Thursday.)

@bschnurr, it looks like you closed this bug after publishing 2021.12.1, but this fix was introduced afterward. It's included in Pyright 1.1.195.

**Comment by bschnurr** [comment]
Sorry about that. New Pylance version 2021.12.2 was just released.
You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021122-13-december-2021)
